tests/acceptance/apps_dns/apps_dns_test.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2020 SUSE
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apps_dns_test
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"net/http"
+
+	"github.com/miekg/dns"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/SUSE/kubecf-apps-dns/tests/acceptance/apps_dns/fake"
+)
+
+var _ = Describe("AppsDns", func() {
+	It("should return a DNS Server Failure when the Service Discovery Controller fails", func() {
+		// Prepare
+		domainName := dns.Fqdn("example.com")
+		sdc.Handle(domainName, func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusInternalServerError)
+		})
+
+		// Assert
+		By("sending a question type A")
+		res, err := dnsQuery(domainName, dns.TypeA)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Rcode).To(Equal(dns.RcodeServerFailure))
+		Expect(res.Answer).To(BeEmpty())
+
+		By("sending a question type AAAA")
+		res, err = dnsQuery(domainName, dns.TypeAAAA)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Rcode).To(Equal(dns.RcodeServerFailure))
+		Expect(res.Answer).To(BeEmpty())
+	})
+
+	It("should handle other question types other than A and AAAA", func() {
+		// Assert
+		By("performing a reverse DNS lookup for 1.1.1.1")
+		res, err := dnsQuery("1.1.1.1.in-addr.arpa.", dns.TypePTR)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+		Expect(res.Answer).ToNot(BeEmpty())
+		Expect(res.Answer[0].(*dns.PTR).Ptr).To(Equal("one.one.one.one."))
```

Comment:
Hmm, that means this test requires real internet access. Probably okay for now, but might come back medium-to-long term to bite us.

Reply:
Yes, it does, like the one we check for `github.com` (even though I expect github to respond before it reaches a DNS outside its network). I don't think this is a big deal even long-term. If it becomes, this test is simple enough anyone could change it to point to something else.